README.md

```diff
@@ -75,85 +75,107 @@ Vue.component('AsyncComponent', (resolve, reject) => {
 ## :gear: Configs
 
 This plugin provides two predefined configs:
-- `plugin:vue/base` - contains necessary settings for this plugin to work properly
-- `plugin:vue/recommended` - extends base config with recommended rules (the ones with check mark :white_check_mark: in the table below)
+- `plugin:vue/base` - Settings and rules to enable correct ESLint parsing
+- `plugin:vue/essential` - Above, plus rules to prevent errors or unintended behavior
```

Comment:
Those configs seems very nice, but I'm not sure we should group them like this in readme. Current groups are common in eslint ecosystem (take a look at original rules https://eslint.org/docs/rules/). Plus later we will probably have some rules that we don't want to enable in any config and give the user full freedom in deciding whether he needs it or not. With the proposed granulation we would also have to add another category - which may lead to unnecessary churn. Also I'm not sure whether maintaining 4 configs is a way to go. I'd rather keep it  simple stupid and provide only `base` and `recommended` config. What do you think @chrisvfritz @mysticatea ?

Reply:
Those configs seems very nice, but I'm not sure we should group them like this in readme. Current groups are common in eslint ecosystem (take a look at original rules eslint.org/docs/rules). Plus later we will probably have some rules that we don't want to enable in any config and give the user full freedom in deciding whether he needs it or not. With the proposed granulation we would also have to add another category - which may lead to unnecessary churn. Also I'm not sure whether maintaining 4 configs is a way to go. I'd rather keep it simple stupid and provide only base and recommended config. What do you think @chrisvfritz @mysticatea ?

===

> Current groups are common in eslint ecosystem (take a look at original rules eslint.org/docs/rules).

I agree it makes sense to stick with ESLint conventions wherever there's not a compelling reason to stray, but I did see some compelling reasons here.

First, I'm thinking that most people will probably discover the ESLint plugin through the style guide, so I saw a strong benefit in not forcing users to context switch after arriving here. If they want to enforce priority A rules, they'll know within seconds exactly what to do. There will be no wondering, "Wait, is Possible Errors what I want to enforce Priority A rules?"

For users who have never seen the style guide, but are familiar with ESLint conventions, I don't think the new name will cause any confusion - but perhaps there's something I'm not foreseeing. What do you think?

Second, I think there's value in the increased granularity. "Stylistic Issues" is too broad in my opinion, which is why I split stylistic rules between Priority B and C. For Priority B, there's one or more possible conventions that offer clear benefits over the alternatives. For Priority C, there are really many good options and the primary value is just in picking something and sticking to it.

Finally, I moved `jsx-uses-vars` out of "Variables" because I just don't find that a useful category, despite ESLint's recommendation. Their rules in that category include a mix of possible errors (e.g. `no-use-before-define`) and stylistic issues (e.g. `init-declarations`) and this is the only language feature (apart from ES6) that receives special treatment. I also feel that `jsx-uses-vars` only _incidentally_ has to do with variables - it's really a parsing extension that allows ESLint to work correctly, which is why I added it to `base`.

> Plus later we will probably have some rules that we don't want to enable in any config and give the user full freedom in deciding whether he needs it or not.

I'm not sure I understand this one. Don't users have the same amount of freedom either way, since they can override whatever they want? In terms of ease of use, I can't think of a case where more nested rule sets would worsen the user experience or make it more difficult for them to do something. Am I missing something though?

> With the proposed granulation we would also have to add another category - which may lead to unnecessary churn.

I might need more context to understand this one, because adding a new category wouldn't be a lot more code nor increase its complexity, if I'm understanding correctly. Can you provide a specific example of what you're thinking might happen?

> Also I'm not sure whether maintaining 4 configs is a way to go. I'd rather keep it simple stupid and provide only base and recommended config.

This is another one where I might need more context. The configs are generated with a script, so I don't believes there's any extra maintenance needed.

Reply:
Okay, I get your point better now @chrisvfritz Thanks for adding more context :)

And regarding last three paragraphs that you quoted:
I was thinking about the case where we want to introduce a rule, but we don't want to enable it in any config by default. Let's say we'd want to give people possibility to use the given rule, but it's not aligned with the official styleguide for some reason.
Maybe it's an edge case, but we had a similar scenario in eslint-plugin-ember.

Adding another group is not a big deal though, so I think that we can move forward with the current state, and iterate over it later on 👍

Reply:
@michalsnik I agree it'd be good to introduce a category like that when the need arises - and I can already think of use cases for it. For example, I like to disallow filters in many projects, because they add one more thing that developers have to learn without enabling them to do anything new. That kind of thing wouldn't fit into any (existing) categories of the style guide though.